tests/Handler/StreamHandlerTest.php

```diff
@@ -201,6 +201,24 @@ public function testAutomaticallyDecompressGzip()
         self::assertTrue(!$response->hasHeader('content-length') || $response->getHeaderLine('content-length') == $response->getBody()->getSize());
     }
 
+    public function testAutomaticallyDecompressGzipHead()
+    {
+        Server::flush();
+        $content = \gzencode('test');
+        Server::enqueue([
+            new Response(200, [
+                'Content-Encoding' => 'gzip',
+                'Content-Length'   => \strlen($content),
+            ], $content)
+        ]);
+        $handler = new StreamHandler();
+        $request = new Request('HEAD', Server::$url);
+        $response = $handler($request, ['decode_content' => true])->wait();
+
+        // Verify that the content-length matches the encoded size.
+        self::assertTrue(!$response->hasHeader('content-length') || $response->getHeaderLine('content-length') == \strlen($content));
```

Comment:
Im honestly not sure. This is just following a pattern. We do the same thing in `testAutomaticallyDecompressGzip` and `testDecodesGzippedResponsesWithHeader`

Reply:
I re-used the implementation of `testAutomaticallyDecompressGzip`. Also most real-world responses with `content-encoding: gzip` will also be `transfer-encoding: chunked`, because the gzip compression is performed in a streaming fashion.

Reply:
I think we should make the tests as strict as possible.

Reply:
I'll leave changing this to a separate PR. Then I can change it on 4(+?) places. 

I'll do it right away.